pkg/neonvm/ipam/ipam.go

```diff
@@ -49,30 +50,38 @@ type Temporary interface {
 
 type IPAM struct {
 	Client
-	Config IPAMConfig
+	Config  IPAMConfig
+	metrics *IPAMMetrics
 
 	mu                 sync.Mutex
 	concurrencyLimiter *semaphore.Weighted
 }
 
+type IPAMParams struct {
+	NadName          string
+	NadNamespace     string
+	ConcurrencyLimit int
+	MetricsReg       prometheus.Registerer
+}
+
 func (i *IPAM) AcquireIP(ctx context.Context, vmName types.NamespacedName) (net.IPNet, error) {
-	ip, err := i.runIPAM(ctx, makeAcquireAction(ctx, vmName))
+	ip, err := i.runIPAMWithMetrics(ctx, makeAcquireAction(ctx, vmName), IPAMAcquire)
```

Comment:
suggestion(non-blocking): Could we de-duplicate the `makeFooAction` + `IPAMFoo` in the args, so that metric label is provided by the action?

(maybe e.g., `IPAMAction` should be a struct or interface, so it can self-describe?)

Reply:
That's fair, let me keep it this way for now, and I'll do that next time I work with this code.

Reply:
Sounds good -- worth opening an issue for it?

Reply:
I'll write it down for now.